core/src/main/clojure/xtdb/expression/temporal.clj

```diff
@@ -583,12 +613,21 @@
               :arg-types [:utf8 :utf8 :int :utf8 :int]}]
     (expr/codegen-call expr)))
 
+(defn duration-from-seconds [^double seconds]
+  ;; apparently EE pases durations as long - so here we do as well
+  (long (* 1000 seconds)))
+
+(defmethod expr/codegen-cast [:f64 :duration] [_expr]
+  {:return-type :duration
```

Comment:
`:duration` isn't a return-type on its own, needs to be `[:duration <time-unit>]` (should be other examples of this in this file). You'll be able to extract the requested time-unit from the `_expr` (again, see other examples in the file).